### DISCUSSION TITLE
Install old version of Nextcloud

### DISCUSSION THREAD

**Discussion by autoantwort** [nextcloud/all-in-one#3739] [Questions] [answered]
I currently have a nextcloud instance (version 23.0.12) on a turnkey container that I can not update. 

I want to migrate the files and the database to not break all the existing links. 

According to the [migration guide](https://github.com/nextcloud/all-in-one/blob/main/migration.md#migrate-the-files-and-the-database) I have to make sure that _your old instance is on exactly the same version like the version used in Nextcloud AIO._

So I have to install Nextcloud AIO in such a way so that I get version `23.0.12` because I can not update the turnkey linux container (I can not update nextcloud in the container because the php version is too old and I can not install a new php version in the turnkey container). 
My idea was to install an old tag of the aio container, but which tag contains which version of nextcloud? And it is then possible to incrementally update the master container to the latest tag?

**Comment by szaimen** [comment, marked as answer]
Hi, there is no AIO version that includes Nextcloud 23.0.12. So you will need to update your turnkey container first to a version that AIO has shipped. Best would be 27.1.3 as that is currently included in AIO.

**Reply by autoantwort** [reply]
> So you will need to update your turnkey container first to a version that AIO has shipped.

I wish I could do that, but this is not really supported (The reason why I want to get away from it). Maybe I will find a way. 

I was hoping that 23.0.12 and 23.0.6 are similar enough 😅  Maybe I can downgrade my nextcloud instance.

Thank you for your answer!

**Reply by szaimen** [reply]
> Maybe I can downgrade my nextcloud instance.

I fear this is not possible with Nextcloud.

**Reply by autoantwort** [reply]
I don't know the content of the database, but since there was no database migration between 23.0.6 and 23.0.12 (`git diff --name-only v23.0.6..v23.0.12 | grep Migration` gives no results), I hoped I could try to migrate the database anyway and maybe have luck 😇

**Reply by szaimen** [reply]
There might be other apps that had a migration in between...
